### PR TITLE
Values.yaml support

### PR DESCRIPTION
**Pull request by ishmaelthedestroyer**
### Overview

This PR adds support for a `values.yaml` file for configuring helm charts.

### Changes

- support `values.yaml` file for configuring helm charts
- new `supported_microservices` variable to determine which charts are supported in current Paragon version
- added support for separate Redis and Postgres instances by overriding variables in `values.yaml`

### Unrelated Changes

- upgraded package.json from `1.0.2` -> `2.0.0`
- default monitor version to paragon version if not provided in `.env-tf-paragon`
- upgraded helm provider
    - added support for Terraform `-init` argument when running CLI. Needed for supporting Helm upgrade
- removed `'fs-extra` and `@types/fs-extra` in place of native `fs` module

### Breaking Changes

If migrating from `1.x', you'll need to migrate your `.env-helm` file to the `values.yaml` file. The README has been updated with a new **Migrations** section.

### CODE REVIEW

File: scripts/cli/base.cli.ts [hunk 1]
```diff
@@ -5,11 +5,12 @@ import chalk from 'chalk';
 import * as commander from 'commander';
 import { compareVersions } from 'compare-versions';
 import * as fs from 'fs-extra';
```
Comment (tedparagon) [inline scripts/cli/base.cli.ts:7]: nit: doesn't look like we use any extras here vs the native `fs`

File: scripts/cli/base.cli.ts [hunk 2]
```diff
@@ -233,15 +232,23 @@ credentials "app.terraform.io" {
...
-        subchart: this.getSubchartEnablementOverrides(helmValues.VERSION),
-        global: {
-          env: helmValues,
-        },
-      };
+      const helmValuesPath: string = `${ROOT_DIR}/.secure/values.yaml`;
+      const helmValuesExists: boolean = await fs
+        .stat(helmValuesPath)
+        .then(() => true)
+        .catch(() => false);
```
[the remark this answers is not in the document]
Comment (ishmaelthedestroyer) [reply scripts/cli/base.cli.ts:239]: I went with this to keep from blocking the event-loop, and the `exists` asynchronous method is marked as deprecated. However, we could change it given that the script doesn't operate in high-volume without issues.

File: scripts/cli/base.cli.ts (hunk 1 above)
Comment (ishmaelthedestroyer) [reply scripts/cli/base.cli.ts:7]: It does provide some sugar methods, like `createFile` for creating empty files. I couldn't find a `touch` method in the native `fs` module but I could just write an empty file.

File: scripts/cli/base.cli.ts [hunk 3]
```diff
@@ -260,40 +267,57 @@ credentials "app.terraform.io" {
...
+    return microservices.filter((microservice: Microservice): boolean => {
+      // hades, pheme and plato expected to be added in v2.67.0 in non-GCP environments
+      // delete the charts if version is before that and not in GCP
+      const hasHadesAndPlatoAndPheme: boolean =
+        isLatest || compareVersions(sanitizedParagonVersion, 'v2.67.0') >= 0;
+
+      // filter unused microservices
+      if (
+        !hasHadesAndPlatoAndPheme &&
+        [Microservice.HADES, Microservice.PLATO].includes(microservice)
```
Comment (tedparagon) [inline scripts/cli/base.cli.ts:289]: should `PHEME` be included here also?

Comment (ishmaelthedestroyer) [reply scripts/cli/base.cli.ts:289]: Ah good catch, thank you

**Review by tedparagon — APPROVED** [review]

**Pull request merged**